service/api_logic/user_logic.py

```diff
@@ -138,6 +146,57 @@ async def create_access_token_response(self, user):
         return response_json
 
 
+    def add_preferences(self, dto: UpdateUserPreferencesDTO):
+        new_dto_by_type_of_preference = self.dto_for_type_of_preference(dto)
+
+        existing_sports = self._preferences_dal.get_all_sport_preference_indexes()
+        if dto.type == SPORT_TYPE:
+            valid_preferences = set([sport_id for sport_id in dto.preferences if sport_id in existing_sports])
+
+        if dto.type == TEAM_TYPE:
+            valid_preferences = set([team_id for team_id in dto.preferences])
+
+        if not valid_preferences:
+            raise IncorrectPreferencesError()
+
+        self.add_valid_user_preferences(new_dto_by_type_of_preference, dto, valid_preferences)
+
+        return CommonResponse().to_dict()
+
+
+    def get_user_preferences(self, dto: UpdateUserPreferencesDTO):
+        new_dto_by_type_of_preference = self.dto_for_type_of_preference(dto)
+
+        prefs = self._preferences_dal.get_user_preferences(new_dto_by_type_of_preference, dto)
+        if dto.type == SPORT_TYPE:
+            return OutputSportPreferences(many=True).dump(prefs)
+
+        if dto.type == TEAM_TYPE:
+            return  OutputTeamPreferences(many=True).dump(prefs)
+
+
+    def get_all_sport_preferences(self):
+        return self._sport_dal.get_all_sports()
```

Comment:
We just return all the sports we have. I don't see how "preferences" in the name could be connected to that